src/app.js

```diff
@@ -0,0 +1,28 @@
+var allnews = new ArticleList()
+
+fetch("http://content.guardianapis.com/search?show-fields=all&q=uk&page-size=12&api-key=fc20c2bc-137d-4f7b-a6c3-3c4efb09563d")
+
+.then(function(response) {
+  return response.json();
+})
+
+.then(function(data) {
+    var interface = data.response.results
+
+    for(var i = 0; i < interface.length; i++) {
+      allnews.save(interface[i].webTitle, interface[i].fields.body, interface[i].webUrl, interface[i].webPublicationDate, interface[i].fields.thumbnail, interface[i].fields.main)
+    }
+
+    buildGrid(allnews._articles);
+
+    var hash = getIDFromUrl();
+    if(hash) {
+        viewSingleArticle(allnews, hash);
+    }
+
+    hashChangeEventListener(allnews);
+  })
```

Comment:
It feels like you are doing a lot in this file - first you fetch the data, then you create a list of models, then you build a grid, then you check the hash, then you add the event listener. It feels like there are a lot of responsibilities here. 

Fetching the data feels like it belongs in an API service class, building the models and returning a view feels like it is part of the controller, interpreting the hash belongs in a router. Setting up the hash event listener and building the controllers (i.e. application setup) feel like the only things that we should really have in an app.js in this example.